src/Controller/HybridAuthController.php

```diff
@@ -147,22 +160,39 @@ protected function authenticateUser($account_data) {
    * Callback path for HybridAuth.
    */
   protected function getEndpointPath() {
-    return $this->urlGenerator->generateFromRoute('hybridauth.endpoint', [], ['absolute' => TRUE]);
+    return $this->urlGenerator->generateFromRoute(
+        'hybridauth.endpoint', [], ['absolute' => TRUE]
+    );
   }
 
   /**
    * Gets providers configuration.
+   *
+   * @param string $provider_id
+   *
+   * @return array
    */
-  protected function getConfiguration() {
-    // @todo add admin pages and get key & secret from config.
-    // @todo it works with Linkedin only now - need to make more flexible.
+  protected function getConfiguration($provider_id = '') {
```

Comment:
for me having getConfiguration by provider id in AuthController looks weird. We need to find out better solution

Reply:
@ApacheEx Are a new service will be good?

Reply:
yeah, I think drupal service makes sense

Reply:
Okey. Added services for getConfiguration of the provider.